single_number.py

```diff
@@ -0,0 +1,34 @@
+# Problem: Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.
+# You must implement a solution with a linear runtime complexity and use only constant extra space.
+from collections import defaultdict
```

Comment:
According to Mo, avoid statements like `from foo import bar`. It's better to use the fully qualified function name like `foo.bar` so it's easier to read.